docker-compose.yml

```diff
@@ -81,4 +81,52 @@ services:
         condition: service_started
 
   redis:
-    image: redis:7.4.5@sha256:b52fce7a50c718c6836eef354f19428a9c9a6802379da6eb8a23853152e23f89
\ No newline at end of file
+    image: redis:7.4.5@sha256:b52fce7a50c718c6836eef354f19428a9c9a6802379da6eb8a23853152e23f89
+
+  keycloak:
+    image: quay.io/keycloak/keycloak:25.0.6@sha256:82c5b7a110456dbd42b86ea572e728878549954cc8bd03cd65410d75328095d2
+    depends_on:
+      keycloak-database:
+        condition: service_healthy
+    ports:
+      - "8081:8080"
+      - "9001:9000"
+    healthcheck:
+      test: [ "CMD-SHELL", "exec 3<>/dev/tcp/127.0.0.1/9000;echo -e 'GET /health/ready HTTP/1.1\r\nhost: http://localhost\r\nConnection: close\r\n\r\n' >&3;if [ $? -eq 0 ]; then echo 'Healthcheck Successful';exit 0;else echo 'Healthcheck Failed';exit 1;fi;" ]
+      interval: 10s
+      timeout: 10s
+      retries: 10
+    environment:
+      - KEYCLOAK_ADMIN=admin
+      - KEYCLOAK_ADMIN_PASSWORD=admin
+      - DB_VENDOR=postgres
+      - DB_ADDR=keycloak-database
+      - DB_USER=keycloak
+      - DB_NAME=keycloak
+      - DB_PASSWORD=keycloak
+    volumes:
+      - ./scripts/docker-compose/imports/keycloak/realms:/opt/keycloak/data/import
```

Comment:
Where is the realm(s) to import though? I don't see those in this PR.

Introducing Keycloak is needed for the user authentication part? Is this really really needed? Because adding Keycloak will introduce a lot of complexity. This is only the start of it. For sure you will then also need to automatically import a realm. The question is also: which realm? Not the ZAC realm I guess because the Office Add-in is not dependent on ZAC but which realm then? I have a feeling this needs a bit more thought?

Reply:
If this is just a start without any realm (i.e. only the default Keycloak master realm) then this volume mapping should be removed for now.